### PR TITLE
feat: authenticate with github app installation and repo standardization

### PR DESCRIPTION
**Pull request by jmeridth**
## Proposed Changes

**I'd highly suggest this be a major version bump (3.0.0) as there are breaking changes:**

- [x] standardize boolean environment variables (true or false)
- [x] drop GITHUB_SERVER_URL and instead use GITHUB_ENTERPRISE_URL for enterprise customers (matches our other repos)

Willing to go the deprecation route but this is the simpler path.

GitHub App Installation:

- [x] add ability to authenticate with GitHub App Installation
- [x] add tests for GitHub App Installation
- [x] add more tests for env vars

Repo updates to match standards:

- [x] alphabetize :allthethings:
- [x] add requirements-test.txt to version our testing dependencies
- [x] add superlinter linting action
- [x] add linter configs
- [x] update Makefile to point linting commands to our linter configs that the GitHub Action workflows use
- [x] add checkov exemptions to Dockerfile
  - custom user - can't do, need root for GitHub action workspace access
  - health check - don't need
- [x] update README, splitting authentication section (will port to other repos)
- [x] change .pylintrc to .python-lint (pylint default config file)
- [x] updated all examples to default to read contents permissions
- [x] standardize the pull_request_template.md

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer
- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by jmeridth** [comment]
@zkoppert yes, I agree, this is HUGE.  I feel a "rip the band aid off" moment allows for it.  There is still an opportunity to split the auth and env stuff out like we do in cleanowners but this gets us in right direction.  If you want me to split repo standardization and GitHub App Installation auth, please let me know.

**Comment by jmeridth** [comment]
The size of this change is daunting.  There are 3 whole new files that are contributing:

- .github/workflows/linter.py
- test_config.py
- docs/authenticating-with-github-app-installation.md

**Comment by jmeridth** [comment]
Whoa.  Looking into the issues.

File: README.md [hunk 1]
```diff
@@ -45,13 +45,15 @@ on:
...
-  pull-requests: read
+  contents: read
 
 jobs:
   build:
     name: issue metrics
     runs-on: ubuntu-latest
+    permissions:
+      issues: write
+        pull-requests: read
```
Comment (zkoppert) [inline README.md:56]: alignment looks off here
```suggestion
      issues: write
      pull-requests: read
```

**Comment by jmeridth** [comment]
iterating through fixes.  there are a decent amount.

**Comment by zkoppert** [comment]
👀

File: config.py [hunk 2]
```diff
@@ -8,86 +8,177 @@
...
+        return None
+
+
+def get_env_vars(test: bool = False) -> EnvVars:
     """
     Get the environment variables for use in the script.
 
     Returns EnvVars object with all environment variables
     """
+    if not test:
```
Comment (zkoppert) [inline config.py:125]: Gold!

**Review by zkoppert — APPROVED** [review]
🚀

**Pull request merged**